tests/system/demos/horovod/test_horovod.py

```diff
@@ -0,0 +1,78 @@
+import pathlib
+
+import mlrun
+
+from tests.system.base import TestMLRunSystem
+from tests.system.demos.base import TestDemo
+
+
+@TestMLRunSystem.skip_test_if_env_not_configured
+class TestHorovodTSv2(TestDemo):
+
+    project_name = 'horovod-project'
+
+    def create_demo_project(self) -> mlrun.projects.MlrunProject:
+        self._logger.debug('Creating horovod project')
+        demo_project = mlrun.new_project(
+            self.project_name, str(self.assets_path), init_git=True
+        )
+
+        mlrun.mount_v3io()
+
+        self._logger.debug('Uploading training file')
+        trainer_src_path = str(self.assets_path / 'horovod_training.py')
+        trainer_dest_path = pathlib.Path('/assets/horovod_training.py')
+        stores = mlrun.datastore.store_manager.set()
+        datastore, subpath = stores.get_or_create_store(
+            self._get_v3io_user_store_path(trainer_dest_path)
+        )
+        datastore.upload(subpath, trainer_src_path)
```

Comment:
That part looks like 💩  - not your fault of course, you're just using the existing SDK functions,
I honestly don't understand what each line is doing here, if you have a vision for how to make the data stores SDK more clear - trello it

Reply:
Yeah I know, that is exactly why it took me over a day to figure out how to use it.
I'll think of something